manifests/haproxy/autoload.pp

```diff
@@ -36,13 +36,9 @@
             $parts = [$options['_domain_prefix'], $env, $options['_domain_suffix']].filter |$item| { !$item.empty }
             $domain_env = join($parts, '.')
           }
-          file { [$path_env,
-            "$path_env/current",
-            "$path_env/current/redirects"]:
-            ensure => directory,
-            owner => $http_edge_path_owner,
-            replace => false
-          }
+          ensure_resource('file', $path_env, {'ensure' => 'directory', 'owner' => $http_edge_path_owner, 'replace' => 'false' })
```

Comment:
You must do whatever is in your hand to never repeat 3 times the same arguments in function calls.

The evident option is to create a hash with the arguments:

```puppet
$file_args = { ... }
ensure_resource('file', $path_env, $file_args)
ensure_resource('file', "$path_env/current", $file_args)
ensure_resource('file', "$path_env/current/redirects", $file_args)
```

Over this, another not so evident improvement, but clear if you read the documentation is that ensure_resource() accepts a list of titles to create several resources at once. So, in the end, the transformation for the original code is just:

```puppet
ensure_resource('file', [$path_env, "$path_env/current", "$path_env/current/redirects"], {
  ensure => directory,
  owner => $http_edge_path_owner,
  replace => false
  }
)
```

Reply:
The code has been improved with the suggestions and testing went well. So I believe this PR can me merged.